zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/RoleDeletedApplier.java

```diff
@@ -37,7 +37,11 @@ public void applyState(final long key, final RoleRecord value) {
     // Remove roles from users if EntityType.USER exists
     final var userEntities = entities.get(EntityType.USER);
     if (userEntities != null) {
-      userEntities.forEach(userKey -> userState.removeRole(userKey, roleKey));
+      userEntities.forEach(
+          userKey ->
+              userState
+                  .getUser(userKey)
+                  .ifPresent(user -> userState.removeRole(user.getUsername(), roleKey)));
     }
     // todo remove entity from mapping state
```

Comment:
🔧 We can remove this code completely. In the `RoleDeleteProcessor` [here](https://github.com/camunda/camunda/blob/a406f791bb017b34f4d20f0c5304f267efb814ac/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/RoleDeleteProcessor.java#L107-L124), we emmit `RoleIntent.ENTITY_REMOVED` events for each member entity, so the `RoleEntityRemovedApplier` is already invoked to remove these entries from the `userState`.

Reply:
Intersting, I didn't know that. I wonder if we should remove it from the applier, or from the processor. I worry that if a role has a lot of users we could run into maximum batch size problems. 

On the other hand, it'd remove the user state from the applier which is quite nice. Especially since we can also remove the authorization state from it soon, since we can just write an authorization delete command.

Reply:
I think not having it in the applier is nicer. I'll remove it from there. If we run into batch size issues we'll deal with it then. There's an easy workaround, which is to remove a bunch of users manually before deleting the role.

Reply:
We shouldn't have removed it from the processor. That way, the exporter won't be able to clean up the ES/OS entries since we don't read from the ES/OS state, we just append to it. We can discuss this in more detail if you like.